x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/scroll/ExtractedField.java

```diff
@@ -103,7 +103,13 @@ public static ExtractedField newField(String alias, String name, ExtractionMetho
             if (value.length != 1) {
                 return value;
             }
-            value[0] = ((BaseDateTime) value[0]).getMillis();
+            if (value[0] instanceof BaseDateTime) { // script field
```

Comment:
Could we swap the order of those `instanceof` checks? The time value will be coming from a doc_value way more often than from a script field, so I'd rather do the most useful check first.